### ISSUE TITLE
checkstyle plugin inappropriately depends on classes

### ISSUE THREAD

**Issue by xenoterracide** [gradle/gradle#28477]
### Expected Behavior

checkstyle checks should run incrementally and on source only. [Checkstyle is a single source file only processor](https://checkstyle.org/#Limitations). 

I don't know how easy it would be to have checkstyle checks be incremental (meaning only check a file if it's been changed). However, it shouldn't be hard to not have it depend on `classes`

### Current Behavior (optional)

checkstyle plugin waits for compilation to be complete to run checks

### Context

slower verification of checkstyle fixes on things like javadoc than desired. I use a fair number of things that slow compilation down, I'd rather not wait on that when I have parallel configured.

https://github.com/gradle/gradle/blob/614c318451d601d305873b099007d847aaeee42c/platforms/jvm/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java#L138

```
------------------------------------------------------------
Gradle 8.6
------------------------------------------------------------

Build time:   2024-02-02 16:47:16 UTC
Revision:     d55c486870a0dc6f6278f53d21381396d0741c6e

Kotlin:       1.9.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          21.0.2 (Eclipse Adoptium 21.0.2+13-LTS)
OS:           Linux 6.6.19-1-MANJARO amd64
```

**Comment by ov7a** [comment]
This issue will be closed as a duplicate of

* https://github.com/gradle/gradle/issues/14227

Please add your use case and 👍 to that issue.

If you think our analysis is wrong, please provide us with more detailed information explaining why.
